api/cloud/cloud.go

```diff
@@ -94,6 +94,47 @@ func (c *Client) UserCredentials(user names.UserTag, cloud names.CloudTag) ([]na
 	return tags, nil
 }
 
+// UpdateCloudsCredentials updates clouds credentials content on the controller.
+// Passed in credentials are keyed on the credential tag.
+func (c *Client) UpdateCloudsCredentials(cloudCredentials map[string]jujucloud.Credential) ([]params.UpdateCredentialResult, error) {
```

Comment:
It seems we can have UpdateCredentialsCheckModels() take the single tag passed in and construct a map and call this new bulk function and save a lot of duplicated code?